pydata_sphinx_theme/_templates/switcher.html

```diff
@@ -0,0 +1,53 @@
+<div class="dropdown">
+    <button type="button" class="btn btn-primary btn-sm navbar-btn dropdown-toggle" id="switcher_button" data-toggle="dropdown">
+        {{ switcher_prefix }}{{ release }}
+        <span class="caret"></span>
+    </button>
+    <div id="switcher" class="dropdown-menu list-group-flush py-0" aria-labelledby="switcher_button">
+    <!-- dropdown will be populated by javascript on page load -->
+    </div>
+</div>
+
+<script type="text/javascript">
+// Function to construct the target URL from the JSON components
+function buildURL(template, entry) {
+    //
+    var parts = template.split(/[\{\}]/);
+    version_ix = parts.indexOf("version");
+    language_ix = parts.indexOf("language");
+    if (version_ix >= 0) {
+        parts[version_ix] = entry.version;
+    }
+    if (language_ix >= 0) {
+        parts[language_ix] = entry.language;
+    }
+    return parts.join("");
+}
+// Function to populate the (release, language) version switcher
+(function () {
+    // get JSON config
+    $.getJSON("{{ switcher_json_url }}", function(data, textStatus, jqXHR) {
+        // create the nodes first (before AJAX calls) to ensure the order is
+        // correct (for now, links will go to doc version homepage)
+        $.each(data, function(index, entry) {
+            entry.url = buildURL("{{ switcher_template_url }}", entry);
+            if (!("name" in entry)) {
+                entry.name = entry.version;
+            }
+            $("#switcher").append(`<a class="list-group-item list-group-item-action py-1" href="${entry.url}">${entry.name}</a>`);
+        });
+        // see if the current page exists in the other versions of the docs
+        $.each(data, function(index, entry) {
+            let filePath = "{{ pagename }}.html";
```

Comment:
I think this will fail if the documentation uses `dirhtml` builder, since the URLs don't contain `.html`.

Reply:
But for a dirhtml you might also don't care about a working dropdown? (not very familiar with dirhtml though)

Reply:
Yeah, I suppose you will still want a working dropdown. Or maybe, I'm not understanding why you wouldn't 😄 . `dirhtml` works in the same way as `html` builder; the only difference is that instead of using `path/to/page.html` it will create `path/to/page/index.html` and then the URL will be `path/to/page/`

See https://www.sphinx-doc.org/en/master/usage/builders/index.html#sphinx.builders.dirhtml.DirectoryHTMLBuilder for more info

Reply:
OK, I just didn't really know what `dirhtml` was (I was confusing it with a _zipped_ dir, which you can provide as a download option, and that is typically to use locally, in which case you probably don't care about version dropdowns, unless you download the docs for different versions :))

But indeed, for `dirhtml` it would be nice to have it working (no idea how often it is used).